### PR TITLE
Part 3 of Restructuring BeaconBlock Proto - Remove Special Record

### PR DESCRIPTION
**Pull request by terenc3t**
Part of #781

This PR removes special record from beacon block

### CODE REVIEW

File: proto/beacon/p2p/v1/types.proto [hunk 1]
```diff
@@ -156,26 +156,30 @@ message ShardAndCommittee {
...
-  bytes state_root_hash32 = 5;
-  // TODO(781): attestations should be a list of attestionRecords according to spec
-  // at ethereum/eth2.0-specs@11d4473. Update this field to AttestionRecord
-  // and remove the AggregatedAttestion message.
-  repeated AggregatedAttestation attestations = 6 [deprecated=true];
-  // repeated AttestationRecord attestations = 6;
-  // Specials consist of exist, penalities, etc.
-  repeated SpecialRecord specials = 7  [deprecated=true]; // Deprecated in favor of unify specials object w/ attestations.
-  repeated bytes proposer_signature = 8; // Type of [uint384]?
+  repeated bytes ancestor_hash32s = 2;
```
Comment (rauljordan) [inline proto/beacon/p2p/v1/types.proto]: Is this a typo?

File: proto/beacon/p2p/v1/types.proto [hunk 2]
```diff
@@ -156,26 +156,30 @@ message ShardAndCommittee {
...
+  bytes state_root_hash32 = 3;
+  bytes randao_reveal_hash32 = 4;
+  bytes candidate_pow_receipt_root_hash32 = 5;
+  repeated bytes signature = 6; // Type of [uint384]?
 
+  // Block Body
+  BeaconBlockBody body =79;
```
Comment (rauljordan) [inline proto/beacon/p2p/v1/types.proto]: Spacing

File: proto/beacon/p2p/v1/types.proto (hunk 1 above)
Comment (terenc3t) [reply proto/beacon/p2p/v1/types.proto]: I think 32s implies there is a multiple of ancestor hashes because `repeated`

Comment (prestonvanloon) [reply proto/beacon/p2p/v1/types.proto]: No type here. Hash32s is plural

File: proto/beacon/p2p/v1/types.proto [hunk 3]
```diff
@@ -156,27 +156,22 @@ message ShardAndCommittee {
...
-  repeated bytes signature = 8; // Type of [uint384]?
+  repeated bytes ancestor_hash32s = 2;
+  bytes state_root_hash32 = 3;
+  bytes randao_reveal_hash32 = 4;
+  bytes candidate_pow_receipt_root_hash32 = 5;
+  repeated bytes signature = 6; // Type of [uint384]?
+
   // Block Body
-  BeaconBlockBody body = 9;
+  BeaconBlockBody body = 1000;
```
Comment (prestonvanloon) [inline proto/beacon/p2p/v1/types.proto]: Why 1000? Do we expect a lot of fields in this message?

File: proto/beacon/p2p/v1/types.proto [hunk 4]
```diff
@@ -156,27 +156,22 @@ message ShardAndCommittee {
...
-  // TODO(781): attestations should be a list of attestionRecords according to spec
-  // at ethereum/eth2.0-specs@11d4473. Update this field to AttestionRecord
-  // and remove the AggregatedAttestion message.
-  repeated AggregatedAttestation attestations = 6 [deprecated=true];
-  // repeated AttestationRecord attestations = 6;
-  // Specials consist of exist, penalities, etc.
-  repeated SpecialRecord specials = 7  [deprecated=true]; // Deprecated in favor of unify specials object w/ attestations.
-  repeated bytes signature = 8; // Type of [uint384]?
+  repeated bytes ancestor_hash32s = 2;
+  bytes state_root_hash32 = 3;
```
Comment (prestonvanloon) [inline proto/beacon/p2p/v1/types.proto]: FYI, once we launch a v1, we can never change the IDs. It's OK for now though

I'll be sure to document this, but read [this](https://developers.google.com/protocol-buffers/docs/proto#updating) for your FYI.

Comment (terenc3t) [reply proto/beacon/p2p/v1/types.proto]: Thanks. Reading it now

File: proto/beacon/p2p/v1/types.proto (hunk 3 above)
Comment (terenc3t) [reply proto/beacon/p2p/v1/types.proto]: Probably not.. I'll change it back to 9

**Pull request merged**